### ISSUE TITLE
Collecting Statistics from Users

### ISSUE THREAD

**Issue by AndreyNikiforov** [icloud-photos-downloader/icloud_photos_downloader#190]
To better understand how our software is used, it would be very beneficial to collect analytics from the field. Based on that data we'll decide what features can be sunset and which worth more support.

Is there a free service for open source software that support that kind of analytics?

Example:
- collect runtime versions, avg download speed
- analytics outcome: nobody uses python lower than 3.8 and avg download speed 100M
- possible decisions: 
    - drop python prior 3.8
    - low priority for partial downloads work

**Comment by menkej** [comment]
I can understand the intention but I have a bad feeling about programs that "call home" in any way...

I suggest talking about a roadmap regarding supported python versions and make priority of work connected to the things where we have the most "activity" or request by users. Not everything someone would like makes sense. We should focus on the stuff we're using and if someone really wants something be open for pull requests - or, if it interests one of us - lend a hand...

**Comment by AndreyNikiforov** [comment]
> I can understand the intention but I have a bad feeling about programs that "call home" in any way...

Yes, it can be a sensitive subject. Being open about intend and transparent about implementation may help. This functionality may be opt-in or opt-out.

> I suggest talking about a roadmap regarding supported python versions and make priority of work connected to the things where we have the most "activity" or request by users. Not everything someone would like makes sense. We should focus on the stuff we're using and if someone really wants something be open for pull requests - or, if it interests one of us - lend a hand...

"Most activity" is the challenge. So far we can only judge by the number of issues, which shows problem areas out of used.

Let's keep ideas and discussion going.

**Comment by ndbroadbent** [comment]
I also have mixed feelings about it, but I know [Homebrew](https://brew.sh/) has some analytics: https://docs.brew.sh/Analytics

It looks like they just use Google Analytics. There are some privacy-focused analytics companies out there like [Plausible](https://plausible.io/), but I think they don't support events for this kind of use-case (just page views).

I think crash reporting with something like Sentry could also be extremely valuable

**Comment by AndreyNikiforov** [comment]
I looked at homebrew -- nice note explaining `what` and `why`. Has controls to turn analytics on/off. Opt-in on first run.

Thinking of the pattern that may resonate with icloudpd users well:
- opt-in on first run; short motivation/explanation
- documented and easily accessible controls
- transparency 1 -- show data that was sent and when
- transparency 2 -- show results from analytics (on the web site)
